src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -856,12 +856,30 @@ private void GeneratePropertySetter<T>(QualifiedType returnType, T decl, Class @
                 WriteStartBraceIndent();
 
                 var marshal = new CSharpMarshalManagedToNativePrinter(ctx);
-                ctx.ReturnVarName = string.Format("{0}{1}{2}",
+                
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    string arrPtrIden = Helpers.SafeIdentifier("__arrPtr");
```

Comment:
This works too but what I meant was that you could just do:

Helpers.SafeIdentifier("arrPtr")

The point is that SafeIdentifier prepends "__" for you, that's the point. No problem if you leave it this way, I just wanted you to know what the convenience of SafeIdentifier.